hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -1403,16 +1416,35 @@ private JobTaskItemByHostMap generateTaskCountByHost(List<HostState> hosts, Iter
         return rv;
     }
 
-    private List<JobTaskMoveAssignment> pushTasksOntoHost(HostState host, Iterable<HostState> heavyHosts) {
+    private List<JobTaskMoveAssignment> pushTasksOntoHost(HostState host, Collection<HostState> heavyHosts) {
+        int moveLimit = config.getTasksMovedFullRebalance();
         MoveAssignmentList moveAssignments = new MoveAssignmentList(spawn, taskSizer);
+
+        List<HostState> lightHostList = Arrays.asList(host);
+        double byteLimitFactor = 1;
+
         for (HostState heavyHost : heavyHosts) {
-            double byteLimitFactor =
-                    1 - ((double) moveAssignments.getBytesUsed() / config.getBytesMovedFullRebalance());
-            moveAssignments.addAll(pushTasksOffHost(heavyHost, Collections.singletonList(host), true, byteLimitFactor,
-                                                    config.getTasksMovedFullRebalance(), true));
+            List<JobTaskMoveAssignment> assignments =
+                    getMoveAssignments(heavyHost, lightHostList, true, byteLimitFactor, moveLimit, true);
```

Comment:
If you use `1` as the `moveLimit` for `getMoveAssignments`, it would be clearer that you are only moving at most one task from each `heaveHost`. Either way, some comments would help others understand what the code does.

Reply:
Retained moveLimit with the original value to allow other moveAssignments to be added if the 1st one is rejected and added javadoc to explain what the method does.

Reply:
Add a comment on your for loop that specifically states that only one assignment will be added.